addons/calendar_sms/models/calendar.py

```diff
@@ -40,33 +43,10 @@ class AlarmManager(models.AbstractModel):
     _inherit = 'calendar.alarm_manager'
 
     @api.model
-    def get_next_mail(self):
+    def _send_reminder_email(self, partners=None):
         """ Cron method, overridden here to send SMS reminders as well
         """
-        result = super(AlarmManager, self).get_next_mail()
-
-        cron = self.env.ref('calendar.ir_cron_scheduler_alarm', raise_if_not_found=False)
-        if not cron:
-            # Like the super method, do nothing if cron doesn't exist anymore
-            return result
-
-        now = fields.Datetime.to_string(fields.Datetime.now())
-        last_sms_cron = cron.lastcall
-
-        interval_to_second = {
-            "weeks": 7 * 24 * 60 * 60,
-            "days": 24 * 60 * 60,
-            "hours": 60 * 60,
-            "minutes": 60,
-            "seconds": 1
-        }
-
-        cron_interval = cron.interval_number * interval_to_second[cron.interval_type]
-        events_data = self._get_next_potential_limit_alarm('sms', seconds=cron_interval)
-
-        for event in self.env['calendar.event'].browse(events_data):
-            max_delta = events_data[event.id]['max_duration']
-            event_start = fields.Datetime.from_string(event.start)
-            for alert in self.do_check_alarm_for_one_date(event_start, event, max_delta, 0, 'sms', after=last_sms_cron, missing=True):
-                event.browse(alert['event_id'])._do_sms_reminder()
-        return result
+        super()._send_reminder_email()
+        events_ids = self._get_alarmed_events_ids(ttype='sms')
+        events = self.env['calendar.event'].search([('id', 'in', events_ids)])
```

Comment:
browse ?
Or even - why not returning browse records in _get_alarmed_events_ids directly ?
And do events.ids when needed ?

Reply:
+1 _get_alarmed_events_ids could return events, like _get_events_to_notify . Or something like that.

Reply:
@dbeguin can't use browse as we don't have anything in cache yet, browse doesn't fetch the records from the database, search does.

Reply:
Maybe there is something I did not understand in this code, but you may browse whatever you want, especially those IDs that come from an sql request and therefore already exist. When accessing it in _do_sms_reminder if anything is not in cache, a read will be performed as far as I know.

Reply:
My bad... both of you are right, one day I'll be able to use that ORM I help coding...